---
title: "Genomic and epigenomic architecture of a plant pathogen"
date: 2020-01-01T11:45:16+01:00

tags:
 - Genomics

summary: "Using several fully assembled *Zymoseptoria* genomes, we describe here a high level of conservation in genomic, epigenomic and transcriptomic composition and structure across the genus *Zymoseptoria*, despite previous reports of high genomic plasticity in *Z. tritici*. The compartmentalized genome allows the maintenance of a functional core genome co-occurring with a highly variable accessory genome."


image:
  preview_only: true
  alt_text: Circos plot from Z.tritici
  focal_point: "Smart"
---

[Link to the publication](https://doi.org/10.1186/s12864-020-06871-w)

## Background
Antagonistic co-evolution can drive rapid adaptation in pathogens and shape genome architecture. Comparative genome analyses of several fungal pathogens revealed highly variable genomes, for many species characterized by specific repeat-rich genome compartments with exceptionally high sequence variability. Dynamic genome structure may enable fast adaptation to host genetics. The wheat pathogen *Zymoseptoria tritici* with its highly variable genome, has emerged as a model organism to study genome evolution of plant pathogens. Here, we compared genomes of *Z. tritici* isolates and of sister species infecting wild grasses to address the evolution of genome composition and structure.

## Results
Using long-read technology, we sequenced and assembled genomes of *Z. ardabiliae*, *Z. brevis*, *Z. pseudotritici* and *Z. passerinii*, together with two isolates of *Z. tritici*. We report a high extent of genome collinearity among *Zymoseptoria* species and high conservation of genomic, transcriptomic and epigenomic signatures of compartmentalization (see image below).

![Circos plots from Z.tritici and Z.pseudotritici, showing compartmentalization at the genomic, transcriptomic and epigenomic levels](/images/Genome_compartment_circos.png)

We identify high gene content variability both within and between species. In addition, such variability is mainly limited to the accessory chromosomes and accessory compartments. Despite strong host specificity and non-overlapping host-range between species, predicted effectors are mainly shared among *Zymoseptoria* species, yet exhibiting a high level of presence-absence polymorphism within *Z. tritici*. Using in planta transcriptomic data from *Z. tritici*, we suggest different roles for the shared orthologs and for the accessory genes during infection of their hosts.

## Conclusion
Despite previous reports of high genomic plasticity in *Z. tritici*, we describe here a high level of conservation in genomic, epigenomic and transcriptomic composition and structure across the genus *Zymoseptoria*. The compartmentalized genome allows the maintenance of a functional core genome co-occurring with a highly variable accessory genome.
